frontend-react/src/components/header/AdminDropdownNav.test.tsx

```diff
@@ -0,0 +1,87 @@
+import { fireEvent, screen } from "@testing-library/react";
+
+import { renderWithRouter } from "../../utils/CustomRenderUtils";
+
+import { AdminDropdownNav } from "./AdminDropdownNav";
+
+class TestLocalStorage {
+    store: Map<string, string | string[]> = new Map();
+
+    constructor(kvPairs?: Array<{ k: string; v: string }>) {
+        if (kvPairs) {
+            kvPairs.forEach(({ k, v }) => this.store.set(k, v));
+        }
+    }
+
+    getItem(key: string) {
+        return this.store.get(key);
+    }
+
+    setItem(key: string, value: string | string[]) {
+        this.store.set(key, value);
+    }
+
+    removeItem(key: string) {
+        this.store.delete(key);
+    }
+
+    clear() {
+        this.store = new Map();
+    }
+}
+
+const mockLocalStorage = new TestLocalStorage();
+jest.mock("../../pages/misc/FeatureFlags", () => ({
+    CheckFeatureFlag: (feature: string) => {
+        return (
```

Comment:
No need to break this out now since that'd mean also writing test coverage for _this_ too, but in case we every run into the need to mock local storage, we'll have this class and mock example ready to go.